compiler/optimizer/LocalAnalysis.cpp

```diff
@@ -714,7 +714,12 @@ bool TR_LocalAnalysisInfo::isCallLike(TR::Node *node) {
 
    if (node->getOpCode().hasSymbolReference())
       {
-      if (node->getSymbolReference()->getSymbol()->isVolatile() || node->getSymbolReference()->getSymbol()->isMethodMetaData())
+      if (node->getSymbolReference()->getSymbol()->isVolatile() || 
+          (node->getSymbolReference()->getSymbol()->isMethodMetaData()
+#if J9_PROJECT_SPECIFIC
```

Comment:
ask the symbol if it's mutable (getSymbol()->isMutable()), have the Eclipse OMR base class implementation return "true" as the most general answer. Your internal-only subclass can override and return false.

That's the basic approach; please feel free to modify as appropriate for this specific scenario.

Specifically, please do not implement an identity-based test (i.e. the "is this the current thread object symbol" question).